mediagrains/gsf.py

```diff
@@ -601,12 +602,14 @@ def decode_file_headers(self):
         except EOFError:
             raise GSFDecodeError("No head block found in file", self.file_data.tell())
 
-    def grains(self, skip_data=False):
+    def grains(self, skip_data=False, load_lazily=False):
         """Generator to get grains from the GSF file. Skips blocks which aren't "grai".
 
         The file_data will be positioned after the `grai` block.
 
         :param skip_data: If True, grain data blocks will be seeked over and only grain headers will be read
+        :param load_lazily: If True, the grains returned will be designed to lazily load data from the underlying stream
+                            only when it is needed. In this case the "skip_data" parameter will be ignored.
```

Comment:
Should we schedule changing this to make `load_lazily=True` the default for a future v2.0 perhaps?

Reply:
Yes I think that would be a sensible point to introduce it - @jamesba do you want to make a pivotal job to do as such.

Reply:
I was thinking that myself, but felt it would be better to discuss that at some future point when we're used to the functionality.